### DISCUSSION TITLE
Squashing multiple changes

### DISCUSSION THREAD

**Discussion by llamafilm** [netboxlabs/netbox-branching#159] [Ideas]
I'm excited about branching + change management, to protect against bad changes.  Here's a workflow I imagine will happen quite often:

1. User creates a branch and creates a bunch of Devices
2. User requests to merge into main
3. Admin rejects it with feedback because of some typos or incorrect naming convention
4. User makes the requested changes
5. Admin merges into main

In this scenario, it would be great if the changelog does not include the original mistakes.  In git terminology, I would like the changes to be _squashed_.

**Comment by mrmrcoleman** [comment]
Adding some context from the [original NetDev chat](https://netdev-community.slack.com/archives/C01P0FRSXRV/p1729785146321359) here. The request is not to squash all changes in a branch into a single change on `main`, which isn't possible, but to squash changes to the same objects so that only the final state is written to `main`.

**Reply by mrmrcoleman** [reply]
@llamafilm This is being tracked here: https://github.com/netboxlabs/netbox-branching/issues/198
